keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -335,7 +335,7 @@ The versions of CNI plugin binaries that must be used for proper dual-stack func
 ### Services
 
 #### Type NodePort, LoadBalancer, ClusterIP
-Services will only be assigned an address from a single address family (either IPv4 or IPv6). The address family for the Service’s cluster IP is determined by setting the field, `.spec.ipFamily`, on that Service. You can only set this field when creating a new Service. Setting the `.spec.ipFamily` field is optional and should only be used if you plan to enable IPv4 and IPv6 Services and Ingresses on your cluster. The configuration of this field not a requirement for egress traffic. The default address family for your cluster is the address family of the first service cluster IP range configured via the `--service-cluster-ip-range` flag to the kube-controller-manager. The `.spec.ipFamily` field can be set to either IPv4 or IPv6.
+Services will only be assigned an address from a single address family (either IPv4 or IPv6). The address family for the Service’s cluster IP is determined by setting the field, `.spec.ipFamily`, on that Service. The User can only set this field when creating a new Service. Setting the `.spec.ipFamily` field is optional and should only be used if the user plan to enable IPv4 and IPv6 Services and Ingresses on your cluster. The configuration of this field not a requirement for egress traffic. The default address family for the cluster is the address family of the first service cluster IP range configured via the `--service-cluster-ip-range` flag to the kube-controller-manager. The `.spec.ipFamily` field can be set to either IPv4 or IPv6. `ipFamily` field is expected to allow mutability under the same conditions `ClusterIP` field allows it. Another field will be introduced in `status.preferredIPFamily`. This field data type is an array of `ipFamily`. The value of this field is the values of `--service-cluster-ip-range`.
```

Comment:
> The address family for the Service’s cluster IP is determined by setting the field, `.spec.ipFamily`, on that Service.

style nit: drop the commas

also maybe "can be determined" rather than "is determined" since the field isn't required and the family will still be determined even if you don't specify it.

> The User can only set

random capital "U"

> if the user plan to enable IPv4 and IPv6 Services and Ingresses on your cluster

plan*s*
*their* cluster
And this sentence is confusing anyway. Does it mean you only need to set if you want *both* IPv4 and IPv6 services in the cluster?

> The configuration of this field not a requirement for egress traffic.

It's a Service field... how would it be required for egress traffic?? (Also missing "is")

> The configuration of this field not a requirement for egress traffic.

mention that this is what gets used if you didn't specify `.spec.ipFamily`

> The `.spec.ipFamily` field can be set to either IPv4 or IPv6.

if that needs to be said it should probably be earlier in the paragraph

> `ipFamily` field is expected to allow mutability under the same conditions `ClusterIP` field allows it.

This directly contradicts the 3rd sentence which says that it can only be set when creating a new service.

> Another field will be introduced in `status.preferredIPFamily`.

Which does what?

I think the whole paragraph needs to be rewritten from scratch now...

> This field data type is an array of `ipFamily`.

`ipFamily` isn't a data type.

> The value of this field is the values of `--service-cluster-ip-range`.

I think you mean something like "the value of this field is the IP families of the ranges passed as `--service-cluster-ip-range`" or something like that (ie, it is not the IP ranges themselves). Also clarify who sets it.

Nothing anywhere seems to explain what `preferredIPFamily` is for.

Reply:
I rewrote this entire paragraph. Please check. Note, `preferredIPFamily` has no value for ClusterIP services it is mostly used for headless services.